delete-orphaned-kube-network-load-balancers.sh

```diff
@@ -33,6 +33,34 @@ total=0
 deleted=0
 verb="should be deleted"
 
+
+#######################################
+# sets up a context if we are runing inside a kube cluster
+#######################################
+in_cluster_setup() {
+    # if we are a pod in kube, this env var should exist
+    if [[ -z "$KUBERNETES_SERVICE_HOST" ]] ; then
+        return
```

Comment:
Is the logic that if we don't have `$KUBERNETES_SERVICE_HOST` then the validation on line 51 doesn't make any sense? If so then we might want to change the name of this function to something with `validate` in the name or just call it something like `bail_if_gke_cluster_makes_no_sense` to capture the imperative nature.

Reply:
yes the env var is how we know we are a pod in a cluster.  So this is really setup_if_we_are_in_cluster.

the real thing is that this actually sets the  KUBE_CONTEXT if we are running in a cluster
the validation is  a safety check as we are effectively ignoring KUBE_CONTEXT if it is provided and we are running as a pod in a cluster.   

so really its detect_and_setup_kube_pod

Reply:
reconciled out of band for any who stumble on this